### ISSUE TITLE
getting logo into invoice template 

### ISSUE THREAD

**Issue by hpeyerl** [kevinpapst/kimai2#1540]
I'm trying to get a logo into one of the invoice templates.  I've poured through the documentation at https://www.kimai.org/documentation/invoices.html and tried a bunch of stuff.

(Always doing a 'bin/console kimai:reload' after every change, then reloading the page on my browser.)

1) I've tried putting files in kimai2/var/invoices/  (a modified company.docx and my own .html.twig file.  
2) I've tried adding my files in templates/invoice/renderer.
3) I've tried modifying the files in templates/invoice/renderer.  If I modify company.docx, I can see some changes.  But if I drag a logo into there that doesn't appear to pick it up.
4) I've added '<img src="{{ asset('images/HME_Logo.png') }}">' to default.html.twig in templates/invoice/renderer and made sure that the file is present in assets/images and permissions/ownership are correct.  

With every attempt, I try to select the appropriate 'Invoice Document' when editing the 'Invoice Template'.  I can never see my custom one in the drop-down.

The behavior doesn't appear to match the documentation.

```
# bin/console kimai:version
Kimai 2 - 1.8 stable (Ayumi) by Kevin Papst and contributors.
```

**Comment by kevinpapst** [comment]
> You can use every document name only once: so having kimai.html.twig and kimai.docx will lead to unpredictable results

Only use `var/invoices/` to store your own templates. And use a new filename, do not use an existing one like `company.docx` (as this already exists).

> I've added '' to default.html.twig in templates/invoice/renderer and made sure that the file is present in assets/images and permissions/ownership are correct

When using an HTML, your images must be placed below the public/ directory. Check the source code of the invoice to see which path is used.

**Comment by hpeyerl** [comment]
```
drwxr-xr-x 3 www-data www-data  4096 Mar 11 07:10 .
drwxrwxr-x 9 www-data www-data  4096 Mar 11 06:16 ..
drwxr-xr-x 8 www-data www-data  4096 Mar 10 06:18 .git
-rw-r--r-- 1 www-data www-data     6 Mar 10 06:18 .gitignore
-rw-r--r-- 1 www-data www-data  1091 Mar 10 06:18 LICENSE
-rw-r--r-- 1 www-data www-data   624 Mar 10 06:18 README.md
-rw-r--r-- 1 www-data www-data   277 Mar 10 06:18 export.csv
-rw-r--r-- 1 www-data www-data 24917 Mar 11 07:09 hme.docx
-rw-r--r-- 1 www-data www-data  6586 Mar 11 06:34 hme.html.twig
-rw-r--r-- 1 www-data www-data 18571 Mar 10 06:18 open-spreadsheet.ods
-rw-r--r-- 1 www-data www-data  5685 Mar 10 06:18 spreadsheet.xlsx
[root@kimai /var/www/html/kimai2/var/invoices git:master]
```

After kimai:reload and page reload:

![Screen Shot 2020-03-11 at 7 11 16 AM](https://user-images.githubusercontent.com/6561217/76420367-a046d780-6367-11ea-8fdc-ab65e91b1292.png)

**Comment by kevinpapst** [comment]
Anything in the logs var/logs/prod.log? Do you have a local.yaml?

**Comment by hpeyerl** [comment]
I created a config/packages/local.yaml just in case it wasn't picking up var/invoices but it didn't change anything:

```
kimai:
        invoice:
                defaults: ~
                documents:
                        - 'var/invoices'
```
The behavior is the same with or without the above local.yaml.

Nothing in error.log and access.log is suitably boring:

```
drwxr-xr-x 2 root     adm     4096 Mar 11 06:28 .
drwxrwxr-x 8 root     syslog  4096 Mar 11 06:28 ..
-rw-r----- 1 www-data adm    37238 Mar 11 07:21 access.log
-rw-r----- 1 www-data adm    81545 Mar 11 06:25 access.log.1
-rw-r----- 1 www-data adm      285 Feb 15 06:50 access.log.10.gz
-rw-r----- 1 www-data adm     1189 Feb 14 14:11 access.log.11.gz
-rw-r----- 1 www-data adm      517 Feb 13 08:14 access.log.12.gz
-rw-r----- 1 www-data adm      805 Feb 12 06:18 access.log.13.gz
-rw-r----- 1 www-data adm      578 Feb  9 08:12 access.log.14.gz
-rw-r----- 1 www-data adm      903 Mar 10 06:13 access.log.2.gz
-rw-r----- 1 www-data adm      522 Mar  4 07:31 access.log.3.gz
-rw-r----- 1 www-data adm      446 Mar  3 04:30 access.log.4.gz
-rw-r----- 1 www-data adm      573 Mar  1 06:49 access.log.5.gz
-rw-r----- 1 www-data adm     2730 Feb 28 06:17 access.log.6.gz
-rw-r----- 1 www-data adm      944 Feb 20 13:29 access.log.7.gz
-rw-r----- 1 www-data adm     1128 Feb 19 04:19 access.log.8.gz
-rw-r----- 1 www-data adm      442 Feb 16 07:02 access.log.9.gz
-rw-r----- 1 www-data adm        0 Nov  4 06:28 error.log
-rw-r----- 1 www-data adm     1520 Nov  3 09:37 error.log.1
-rw-r----- 1 www-data adm      411 Dec 23  2018 error.log.2.gz
[root@kimai /var/log/nginx]
# tail -n 5 access.log
192.168.37.29 - - [11/Mar/2020:07:21:12 -0600] "GET /en/invoice/template/3/delete HTTP/1.1" 302 336 "http://kimai.beer.org/en/invoice/template" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/79.0.3945.130 Safari/537.36 OPR/66.0.3515.44"
192.168.37.29 - - [11/Mar/2020:07:21:12 -0600] "GET /en/invoice/template HTTP/1.1" 200 4770 "http://kimai.beer.org/en/invoice/template" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/79.0.3945.130 Safari/537.36 OPR/66.0.3515.44"
192.168.37.29 - - [11/Mar/2020:07:21:13 -0600] "GET /api/timesheets/recent?size=10 HTTP/1.1" 200 5658 "http://kimai.beer.org/en/invoice/template" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/79.0.3945.130 Safari/537.36 OPR/66.0.3515.44"
192.168.37.29 - - [11/Mar/2020:07:21:14 -0600] "GET /en/invoice/template/create/1 HTTP/1.1" 200 4954 "http://kimai.beer.org/en/invoice/template" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/79.0.3945.130 Safari/537.36 OPR/66.0.3515.44"
192.168.37.29 - - [11/Mar/2020:07:21:15 -0600] "GET /api/timesheets/recent?size=10 HTTP/1.1" 200 5658 "http://kimai.beer.org/en/invoice/template/create/1" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/79.0.3945.130 Safari/537.36 OPR/66.0.3515.44"
[root@kimai /var/log/nginx]
```

**Comment by hpeyerl** [comment]
oh sorry. you said prod.log.

```
[2020-03-10 06:29:17] console.ERROR: Error thrown while running command "reload". Message: "Command "reload" is not defined." {"exception":"[object] (Symfony\\Component\\Console\\Exception\\CommandNotFoundException(code: 0): Command \"reload\" is not defined. at /var/www/html/kimai2/vendor/symfony/console/Application.php:632)","command":"reload","message":"Command \"reload\" is not defined."} []
[2020-03-10 06:29:36] console.ERROR: Error thrown while running command "'kimai:reload'". Message: "Command "kimai:reload" is not defined.  Did you mean one of these?     kimai:create-user     kimai:import-v1     kimai:phpcs     kimai:reset-dev     kimai:test-integration     kimai:test-unit     kimai:version" {"exception":"[object] (Symfony\\Component\\Console\\Exception\\CommandNotFoundException(code: 0): Command \"kimai:reload\" is not defined.\n\nDid you mean one of these?\n    kimai:create-user\n    kimai:import-v1\n    kimai:phpcs\n    kimai:reset-dev\n    kimai:test-integration\n    kimai:test-unit\n    kimai:version at /var/www/html/kimai2/vendor/symfony/console/Application.php:632)","command":"'kimai:reload'","message":"Command \"kimai:reload\" is not defined.\n\nDid you mean one of these?\n    kimai:create-user\n    kimai:import-v1\n    kimai:phpcs\n    kimai:reset-dev\n    kimai:test-integration\n    kimai:test-unit\n    kimai:version"} []
[2020-03-10 06:42:04] php.INFO: User Deprecated: Loading the file "../src/API/" from the global resource directory "/var/www/html/kimai2/src" is deprecated since Symfony 4.4 and will be removed in 5.0. {"exception":"[object] (ErrorException(code: 0): User Deprecated: Loading the file \"../src/API/\" from the global resource directory \"/var/www/html/kimai2/src\" is deprecated since Symfony 4.4 and will be removed in 5.0. at /var/www/html/kimai2/vendor/symfony/http-kernel/Config/FileLocator.php:84)"} []
[2020-03-10 06:42:04] php.INFO: User Deprecated: Loading the file "../src/API/" from the global resource directory "/var/www/html/kimai2/src" is deprecated since Symfony 4.4 and will be removed in 5.0. {"exception":"[object] (ErrorException(code: 0): User Deprecated: Loading the file \"../src/API/\" from the global resource directory \"/var/www/html/kimai2/src\" is deprecated since Symfony 4.4 and will be removed in 5.0. at /var/www/html/kimai2/vendor/symfony/http-kernel/Config/FileLocator.php:84)"} []
[2020-03-10 06:42:04] php.INFO: User Deprecated: The "Symfony\Bundle\TwigBundle\Controller\ExceptionController" class is deprecated since Symfony 4.4, use "Symfony\Component\HttpKernel\Controller\ErrorController" instead. {"exception":"[object] (ErrorException(code: 0): User Deprecated: The \"Symfony\\Bundle\\TwigBundle\\Controller\\ExceptionController\" class is deprecated since Symfony 4.4, use \"Symfony\\Component\\HttpKernel\\Controller\\ErrorController\" instead. at /var/www/html/kimai2/vendor/symfony/twig-bundle/Controller/ExceptionController.php:23)"} []
[2020-03-10 06:42:04] php.INFO: User Deprecated: The "Symfony\Bundle\TwigBundle\Controller\PreviewErrorController" class is deprecated since Symfony 4.4, use the "Symfony\Component\HttpKernel\Controller\ErrorController" instead. {"exception":"[object] (ErrorException(code: 0): User Deprecated: The \"Symfony\\Bundle\\TwigBundle\\Controller\\PreviewErrorController\" class is deprecated since Symfony 4.4, use the \"Symfony\\Component\\HttpKernel\\Controller\\ErrorController\" instead. at /var/www/html/kimai2/vendor/symfony/twig-bundle/Controller/PreviewErrorController.php:18)"} []
[2020-03-10 06:42:07] php.CRITICAL: Uncaught Error: unserialize() expects parameter 1 to be string, array given {"exception":"[object] (TypeError(code: 0): unserialize() expects parameter 1 to be string, array given at /var/www/html/kimai2/vendor/jms/metadata/src/ClassMetadata.php:111)"} []
```
Those logs were yesterday before I realized I was behind versions so I updated to 1.8.

**Comment by kevinpapst** [comment]
Ok. Please delete the local.yaml while we search for the problem. 
You use-case is pre-configured.

And your logs say that `kimai:reload` is not defined, which I find highly disturbing.
What is the exact output of `bin/console kimai:reload`?

**Comment by hpeyerl** [comment]
> Those logs were yesterday before I realized I was behind versions so I updated to 1.8.

when 'kimai:reload' didn't work yesterday, I realized I must need to update from 0.whatever to latest.  After updating to 1.8, kimai:reload worked.  So that's a red-herring.

Here's the output regardless.  

```
# !bin
bin/console kimai:reload

Reloading configurations ...
============================



 [OK] All 58 YAML files contain valid syntax.





 [OK] All 261 XLIFF files contain valid syntax.



 Rebuilding your cache, please be patient ...

 // Clearing the cache for the prod environment with debug false


 [OK] Cache for the "prod" environment (debug=false) was successfully cleared.



 // Warming up the cache for the prod environment with debug false


 [OK] Cache for the "prod" environment (debug=false) was successfully warmed.




 [OK] Kimai config was reloaded
```

**Comment by hpeyerl** [comment]
...and I had already deleted local.yaml.

**Comment by kevinpapst** [comment]
Ah right, different date. Ok. And `git status` says "all up to date" without any notable changes?

**Comment by hpeyerl** [comment]
```
# git status
HEAD detached at 1.8
Untracked files:
  (use "git add <file>..." to include in what will be committed)

	../../assets/images/HME_Logo.png

nothing added to commit but untracked files present (use "git add" to track)
```

**Comment by kevinpapst** [comment]
The image is in the wrong location (for the twig files at least). 
> your images must be placed below the public/ directory

https://www.kimai.org/documentation/invoices.html#twig-templates

Which doesn't solve the problem that hme.* files don't show up. At least they should be visible in the dropdowns...

**Comment by hpeyerl** [comment]
Yup.  I've now moved it.

Just a heads-up, I'll be getting ready to head into the office so I won't be able to experiment further until either late tonight or 22hrs from now.

**Comment by hpeyerl** [comment]
Since I don't really know what reload does, on a lark, I did:

```
mv var/cache/prod var/cache/prod.old
bin/console kimai:reload
sudo chown -R www-data:www-data var/cache
```

No change.

**Comment by kevinpapst** [comment]
Can you please delete the directory var/invoices/ (don't create it afterwards) and then use the upload feature from Kimai at /en/invoice/document_upload with the hme twig file.

**Comment by hpeyerl** [comment]
Running out the door but document_upload produced this error so I'll have to fix php.ini or whatever...

```
2020/03/11 08:28:54 [error] 280#280: *2225 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Allowed memory size of 134217728 bytes exhausted (tried to allocate 20480 bytes) in /home/hpeyerl/kimai2/vendor/symfony/debug/Exception/FlattenException.php on line 299
PHP message: PHP Fatal error:  Allowed memory size of 134217728 bytes exhausted (tried to allocate 32768 bytes) in /home/hpeyerl/kimai2/vendor/symfony/debug/Exception/OutOfMemoryException.php on line 1" while reading response header from upstream, client: 192.168.37.29, server: , request: "GET /en/invoice/document_upload HTTP/1.1", upstream: "fastcgi://unix:/run/php/php7.2-fpm.sock:", host: "kimai.beer.org"
```

**Comment by kevinpapst** [comment]
Please check the permissions of the log directory first (use Doctor > File permissions).
This error happens as well, if Kimai cannot write the log file.

**Comment by hpeyerl** [comment]
Doctor->File Permissions is all green checkmarks.  Are we sure PHP should get >128MB?

**Comment by hpeyerl** [comment]
Something else is wrong.  512MB fails in the same way.

Maybe this will help:

```
[2020-03-12 05:58:21] request.CRITICAL: Uncaught PHP Exception Symfony\Component\Security\Core\Exception\AuthenticationCredentialsNotFoundException: "The token storage contains no authentication token. One possible reason may be that there is no firewall configured for this URL." at /home/hpeyerl/kimai2/vendor/symfony/security-core/Authorization/AuthorizationChecker.php line 49 {"exception":"[object] (Symfony\\Component\\Security\\Core\\Exception\\AuthenticationCredentialsNotFoundException(code: 0): The token storage contains no authentication token. One possible reason may be that there is no firewall configured for this URL. at /home/hpeyerl/kimai2/vendor/symfony/security-core/Authorization/AuthorizationChecker.php:49)"} []
[2020-03-12 05:58:21] request.CRITICAL: Uncaught PHP Exception Symfony\Component\Security\Core\Exception\AuthenticationCredentialsNotFoundException: "The token storage contains no authentication token. One possible reason may be that there is no firewall configured for this URL." at /home/hpeyerl/kimai2/vendor/symfony/security-core/Authorization/AuthorizationChecker.php line 49 {"exception":"[object] (Symfony\\Component\\Security\\Core\\Exception\\AuthenticationCredentialsNotFoundException(code: 0): The token storage contains no authentication token. One possible reason may be that there is no firewall configured for this URL. at /home/hpeyerl/kimai2/vendor/symfony/security-core/Authorization/AuthorizationChecker.php:49)"} []
[2020-03-12 05:58:21] request.CRITICAL: Uncaught PHP Exception Symfony\Component\Security\Core\Exception\AuthenticationCredentialsNotFoundException: "The token storage contains no authentication token. One possible reason may be that there is no firewall configured for this URL." at /home/hpeyerl/kimai2/vendor/symfony/security-core/Authorization/AuthorizationChecker.php line 49 {"exception":"[object] (Symfony\\Component\\Security\\Core\\Exception\\AuthenticationCredentialsNotFoundException(code: 0): The token storage contains no authentication token. One possible reason may be that there is no firewall configured for this URL. at /home/hpeyerl/kimai2/vendor/symfony/security-core/Authorization/AuthorizationChecker.php:49)"} []
[2020-03-12 05:58:21] request.CRITICAL: Uncaught PHP Exception Symfony\Component\Security\Core\Exception\AuthenticationCredentialsNotFoundException: "The token storage contains no authentication token. One possible reason may be that there is no firewall configured for this URL." at /home/hpeyerl/kimai2/vendor/symfony/security-core/Authorization/AuthorizationChecker.php line 49 {"exception":"[object] (Symfony\\Component\\Security\\Core\\Exception\\AuthenticationCredentialsNotFoundException(code: 0): The token storage contains no authentication token. One possible reason may be that there is no firewall configured for this URL. at /home/hpeyerl/kimai2/vendor/symfony/security-core/Authorization/AuthorizationChecker.php:49)"} []
[2020-03-12 05:58:21] request.CRITICAL: Uncaught PHP Exception Symfony\Component\Security\Core\Exception\AuthenticationCredentialsNotFoundException: "The token storage contains no authentication token. One possible reason may be that there is no firewall configured for this URL." at /home/hpeyerl/kimai2/vendor/symfony/security-core/Authorization/AuthorizationChecker.php line 49 {"exception":"[object] (Symfony\\Component\\Security\\Core\\Exception\\AuthenticationCredentialsNotFoundException(code: 0): The token storage contains no authentication token. One possible reason may be that there is no firewall configured for this URL. at /home/hpeyerl/kimai2/vendor/symfony/security-core/Authorization/AuthorizationChecker.php:49)"} []
[2020-03-12 05:58:21] request.CRITICAL: Uncaught PHP Exception Symfony\Component\Security\Core\Exception\AuthenticationCredentialsNotFoundException: "The token storage contains no authentication token. One possible reason may be that there is no firewall configured for this URL." at /home/hpeyerl/kimai2/vendor/symfony/security-core/Authorization/AuthorizationChecker.php line 49 {"exception":"[object] (Symfony\\Component\\Security\\Core\\Exception\\AuthenticationCredentialsNotFoundException(code: 0): The token storage contains no authentication token. One possible reason may be that there is no firewall configured for this URL. at /home/hpeyerl/kimai2/vendor/symfony/security-core/Authorization/AuthorizationChecker.php:49)"} []
[2020-03-12 05:58:21] request.CRITICAL: Uncaught PHP Exception Symfony\Component\Security\Core\Exception\AuthenticationCredentialsNotFoundException: "The token storage contains no authentication token. One possible reason may be that there is no firewall configured for this URL." at /home/hpeyerl/kimai2/vendor/symfony/security-core/Authorization/AuthorizationChecker.php line 49 {"exception":"[object] (Symfony\\Component\\Security\\Core\\Exception\\AuthenticationCredentialsNotFoundException(code: 0): The token storage contains no authentication token. One possible reason may be that there is no firewall configured for this URL. at /home/hpeyerl/kimai2/vendor/symfony/security-core/Authorization/AuthorizationChecker.php:49)"} []
[2020-03-12 05:58:21] request.CRITICAL: Uncaught PHP Exception Symfony\Component\Security\Core\Exception\AuthenticationCredentialsNotFoundException: "The token storage contains no authentication token. One possible reason may be that there is no firewall configured for this URL." at /home/hpeyerl/kimai2/vendor/symfony/security-core/Authorization/AuthorizationChecker.php line 49 {"exception":"[object] (Symfony\\Component\\Security\\Core\\Exception\\AuthenticationCredentialsNotFoundException(code: 0): The token storage contains no authentication token. One possible reason may be that there is no firewall configured for this URL. at /home/hpeyerl/kimai2/vendor/symfony/security-core/Authorization/AuthorizationChecker.php:49)"} []
[2020-03-12 05:58:21] request.CRITICAL: Uncaught PHP Exception Symfony\Component\Security\Core\Exception\AuthenticationCredentialsNotFoundException: "The token storage contains no authentication token. One possible reason may be that there is no firewall configured for this URL." at /home/hpeyerl/kimai2/vendor/symfony/security-core/Authorization/AuthorizationChecker.php line 49 {"exception":"[object] (Symfony\\Component\\Security\\Core\\Exception\\AuthenticationCredentialsNotFoundException(code: 0): The token storage contains no authentication token. One possible reason may be that there is no firewall configured for this URL. at /home/hpeyerl/kimai2/vendor/symfony/security-core/Authorization/AuthorizationChecker.php:49)"} []
[2020-03-12 05:58:22] request.CRITICAL: Uncaught PHP Exception Symfony\Component\Security\Core\Exception\AuthenticationCredentialsNotFoundException: "The token storage contains no authentication token. One possible reason may be that there is no firewall configured for this URL." at /home/hpeyerl/kimai2/vendor/symfony/security-core/Authorization/AuthorizationChecker.php line 49 {"exception":"[object] (Symfony\\Component\\Security\\Core\\Exception\\AuthenticationCredentialsNotFoundException(code: 0): The token storage contains no authentication token. One possible reason may be that there is no firewall configured for this URL. at /home/hpeyerl/kimai2/vendor/symfony/security-core/Authorization/AuthorizationChecker.php:49)"} []
[2020-03-12 05:58:22] request.CRITICAL: Uncaught PHP Exception Symfony\Component\Security\Core\Exception\AuthenticationCredentialsNotFoundException: "The token storage contains no authentication token. One possible reason may be that there is no firewall configured for this URL." at /home/hpeyerl/kimai2/vendor/symfony/security-core/Authorization/AuthorizationChecker.php line 49 {"exception":"[object] (Symfony\\Component\\Security\\Core\\Exception\\AuthenticationCredentialsNotFoundException(code: 0): The token storage contains no authentication token. One possible reason may be that there is no firewall configured for this URL. at /home/hpeyerl/kimai2/vendor/symfony/security-core/Authorization/AuthorizationChecker.php:49)"} []
[2020-03-12 05:58:22] request.CRITICAL: Uncaught PHP Exception Symfony\Component\Security\Core\Exception\AuthenticationCredentialsNotFoundException: "The token storage contains no authentication token. One possible reason may be that there is no firewall configured for this URL." at /home/hpeyerl/kimai2/vendor/symfony/security-core/Authorization/AuthorizationChecker.php line 49 {"exception":"[object] (Symfony\\Component\\Security\\Core\\Exception\\AuthenticationCredentialsNotFoundException(code: 0): The token storage contains no authentication token. One possible reason may be that there is no firewall configured for this URL. at /home/hpeyerl/kimai2/vendor/symfony/security-core/Authorization/AuthorizationChecker.php:49)"} []
[2020-03-12 05:58:22] request.CRITICAL: Uncaught PHP Exception Symfony\Component\Security\Core\Exception\AuthenticationCredentialsNotFoundException: "The token storage contains no authentication token. One possible reason may be that there is no firewall configured for this URL." at /home/hpeyerl/kimai2/vendor/symfony/security-core/Authorization/AuthorizationChecker.php line 49 {"exception":"[object] (Symfony\\Component\\Security\\Core\\Exception\\AuthenticationCredentialsNotFoundException(code: 0): The token storage contains no authentication token. One possible reason may be that there is no firewall configured for this URL. at /home/hpeyerl/kimai2/vendor/symfony/security-core/Authorization/AuthorizationChecker.php:49)"} []
[2020-03-12 05:58:22] request.CRITICAL: Uncaught PHP Exception Symfony\Component\Security\Core\Exception\AuthenticationCredentialsNotFoundException: "The token storage contains no authentication token. One possible reason may be that there is no firewall configured for this URL." at /home/hpeyerl/kimai2/vendor/symfony/security-core/Authorization/AuthorizationChecker.php line 49 {"exception":"[object] (Symfony\\Component\\Security\\Core\\Exception\\AuthenticationCredentialsNotFoundException(code: 0): The token storage contains no authentication token. One possible reason may be that there is no firewall configured for this URL. at /home/hpeyerl/kimai2/vendor/symfony/security-core/Authorization/AuthorizationChecker.php:49)"} []
[2020-03-12 05:58:22] request.CRITICAL: Uncaught PHP Exception Symfony\Component\Security\Core\Exception\AuthenticationCredentialsNotFoundException: "The token storage contains no authentication token. One possible reason may be that there is no firewall configured for this URL." at /home/hpeyerl/kimai2/vendor/symfony/security-core/Authorization/AuthorizationChecker.php line 49 {"exception":"[object] (Symfony\\Component\\Security\\Core\\Exception\\AuthenticationCredentialsNotFoundException(code: 0): The token storage contains no authentication token. One possible reason may be that there is no firewall configured for this URL. at /home/hpeyerl/kimai2/vendor/symfony/security-core/Authorization/AuthorizationChecker.php:49)"} []
[2020-03-12 05:58:22] request.CRITICAL: Uncaught PHP Exception Symfony\Component\Security\Core\Exception\AuthenticationCredentialsNotFoundException: "The token storage contains no authentication token. One possible reason may be that there is no firewall configured for this URL." at /home/hpeyerl/kimai2/vendor/symfony/security-core/Authorization/AuthorizationChecker.php line 49 {"exception":"[object] (Symfony\\Component\\Security\\Core\\Exception\\AuthenticationCredentialsNotFoundException(code: 0): The token storage contains no authentication token. One possible reason may be that there is no firewall configured for this URL. at /home/hpeyerl/kimai2/vendor/symfony/security-core/Authorization/AuthorizationChecker.php:49)"} []
[2020-03-12 05:58:22] request.CRITICAL: Uncaught PHP Exception Symfony\Component\Security\Core\Exception\AuthenticationCredentialsNotFoundException: "The token storage contains no authentication token. One possible reason may be that there is no firewall configured for this URL." at /home/hpeyerl/kimai2/vendor/symfony/security-core/Authorization/AuthorizationChecker.php line 49 {"exception":"[object] (Symfony\\Component\\Security\\Core\\Exception\\AuthenticationCredentialsNotFoundException(code: 0): The token storage contains no authentication token. One possible reason may be that there is no firewall configured for this URL. at /home/hpeyerl/kimai2/vendor/symfony/security-core/Authorization/AuthorizationChecker.php:49)"} []
[2020-03-12 05:58:22] request.CRITICAL: Uncaught PHP Exception Symfony\Component\Security\Core\Exception\AuthenticationCredentialsNotFoundException: "The token storage contains no authentication token. One possible reason may be that there is no firewall configured for this URL." at /home/hpeyerl/kimai2/vendor/symfony/security-core/Authorization/AuthorizationChecker.php line 49 {"exception":"[object] (Symfony\\Component\\Security\\Core\\Exception\\AuthenticationCredentialsNotFoundException(code: 0): The token storage contains no authentication token. One possible reason may be that there is no firewall configured for this URL. at /home/hpeyerl/kimai2/vendor/symfony/security-core/Authorization/AuthorizationChecker.php:49)"} []
[2020-03-12 05:58:23] request.CRITICAL: Uncaught PHP Exception Symfony\Component\Security\Core\Exception\AuthenticationCredentialsNotFoundException: "The token storage contains no authentication token. One possible reason may be that there is no firewall configured for this URL." at /home/hpeyerl/kimai2/vendor/symfony/security-core/Authorization/AuthorizationChecker.php line 49 {"exception":"[object] (Symfony\\Component\\Security\\Core\\Exception\\AuthenticationCredentialsNotFoundException(code: 0): The token storage contains no authentication token. One possible reason may be that there is no firewall configured for this URL. at /home/hpeyerl/kimai2/vendor/symfony/security-core/Authorization/AuthorizationChecker.php:49)"} []
[2020-03-12 05:58:23] request.CRITICAL: Uncaught PHP Exception Symfony\Component\Security\Core\Exception\AuthenticationCredentialsNotFoundException: "The token storage contains no authentication token. One possible reason may be that there is no firewall configured for this URL." at /home/hpeyerl/kimai2/vendor/symfony/security-core/Authorization/AuthorizationChecker.php line 49 {"exception":"[object] (Symfony\\Component\\Security\\Core\\Exception\\AuthenticationCredentialsNotFoundException(code: 0): The token storage contains no authentication token. One possible reason may be that there is no firewall configured for this URL. at /home/hpeyerl/kimai2/vendor/symfony/security-core/Authorization/AuthorizationChecker.php:49)"} []
[2020-03-12 05:58:23] request.CRITICAL: Uncaught PHP Exception Symfony\Component\Security\Core\Exception\AuthenticationCredentialsNotFoundException: "The token storage contains no authentication token. One possible reason may be that there is no firewall configured for this URL." at /home/hpeyerl/kimai2/vendor/symfony/security-core/Authorization/AuthorizationChecker.php line 49 {"exception":"[object] (Symfony\\Component\\Security\\Core\\Exception\\AuthenticationCredentialsNotFoundException(code: 0): The token storage contains no authentication token. One possible reason may be that there is no firewall configured for this URL. at /home/hpeyerl/kimai2/vendor/symfony/security-core/Authorization/AuthorizationChecker.php:49)"} []
[2020-03-12 05:58:27] request.CRITICAL: Uncaught PHP Exception Symfony\Component\Security\Core\Exception\AuthenticationCredentialsNotFoundException: "The token storage contains no authentication token. One possible reason may be that there is no firewall configured for this URL." at /home/hpeyerl/kimai2/vendor/symfony/security-core/Authorization/AuthorizationChecker.php line 49 {"exception":"[object] (Symfony\\Component\\Security\\Core\\Exception\\AuthenticationCredentialsNotFoundException(code: 0): The token storage contains no authentication token. One possible reason may be that there is no firewall configured for this URL. at /home/hpeyerl/kimai2/vendor/symfony/security-core/Authorization/AuthorizationChecker.php:49)"} []
[2020-03-12 05:58:43] request.CRITICAL: Uncaught PHP Exception Symfony\Component\Security\Core\Exception\AuthenticationCredentialsNotFoundException: "The token storage contains no authentication token. One possible reason may be that there is no firewall configured for this URL." at /home/hpeyerl/kimai2/vendor/symfony/security-core/Authorization/AuthorizationChecker.php line 49 {"exception":"[object] (Symfony\\Component\\Security\\Core\\Exception\\AuthenticationCredentialsNotFoundException(code: 0): The token storage contains no authentication token. One possible reason may be that there is no firewall configured for this URL. at /home/hpeyerl/kimai2/vendor/symfony/security-core/Authorization/AuthorizationChecker.php:49)"} []
[2020-03-12 05:58:44] php.CRITICAL: Fatal Error: Allowed memory size of 536870912 bytes exhausted (tried to allocate 20480 bytes) {"exception":"[object] (Symfony\\Component\\Debug\\Exception\\OutOfMemoryException(code: 0): Error: Allowed memory size of 536870912 bytes exhausted (tried to allocate 20480 bytes) at /home/hpeyerl/kimai2/vendor/symfony/debug/Exception/FlattenException.php:304)"} []
[2020-03-12 05:58:44] request.CRITICAL: Uncaught PHP Exception Symfony\Component\Debug\Exception\OutOfMemoryException: "Error: Allowed memory size of 536870912 bytes exhausted (tried to allocate 20480 bytes)" at /home/hpeyerl/kimai2/vendor/symfony/debug/Exception/FlattenException.php line 304 {"exception":"[object] (Symfony\\Component\\Debug\\Exception\\OutOfMemoryException(code: 0): Error: Allowed memory size of 536870912 bytes exhausted (tried to allocate 20480 bytes) at /home/hpeyerl/kimai2/vendor/symfony/debug/Exception/FlattenException.php:304)"} []
```

**Comment by kevinpapst** [comment]
Not for a simple upload. Only large PDF exports can consume a pretty large amount of RAM.
As the invoice templates are read on demand, you should not even need to flush the cache when adding new ones.

I found problems when using two files with the same name (excluding the extension).
Please delete the var/invoices/ directory, flush cache, open invoice template edit screen, recreate the directory and add only ONE template - the hme twig file. Then reload the edit form again.

And I have no idea what the above problem is. The memory runs out, because of some kind f recursion.

**Comment by hpeyerl** [comment]
sorry. I deleted that because it was a red-herring.  I think I found a problem.  Somehow kimai2 is finding an old installation I had in my homedir.

**Comment by kevinpapst** [comment]
Oh true, `/var/www/html/kimai2/` vs `/home/hpeyerl/kimai2/`. Probably restart PHP, maybe an aggressive bytecode cache setting.

**Comment by hpeyerl** [comment]
ok.  It's working.

Explanation:  a couple years ago I installed kimai2 in /var/www/html (0.8 or so) and found it to not meet my needs.  At the end of last year I decided to try it again so installed it in the same VM but in my homedir.   When I started working on my invoice templates 2 days ago, I updated the instance in /var/www/html to 1.8 and was trying to make my template changes in there but didn't notice that nginx was pointed at the instance in my homedir which was at 1.4.2.  

Since my last post, I cleaned up, updated the instance in my homedir to 1.8 and it all looked different.  Was able to upload my hme.docx using the button (because there wasn't one in 1.4.2 apparently) and produce an invoice that had my logo.    It wouldn't let me upload hme.html.twig because that's not an acceptable filetype.  I'd prefer to use .twig so I'll just edit in-situ.

Sorry.
